pkg/providers/snow/validator.go

```diff
@@ -12,23 +12,39 @@ const (
 	defaultAwsSshKeyName = "eksa-default"
 )
 
-type Validator struct {
+type validator struct {
```

Comment:
I think we can call this `AwsClientValidator` (or something like that)
It follows the `Validator` interface using an aws client as the underlying implementation